data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaSourceCustomConsumer.java

```diff
@@ -215,20 +213,16 @@ void processAcknowledgedOffsets() {
 
         acknowledgedOffsets.forEach(offsets -> {
             offsets.forEach((partition, offsetRange) -> {
-
                 if (getPartitionEpoch(partition) == offsetRange.getEpoch()) {
                     try {
                         int partitionId = partition.partition();
-                        if (!partitionCommitTrackerMap.containsKey(partitionId)) {
-                            OffsetAndMetadata committedOffsetAndMetadata = consumer.committed(partition);
-                            Long committedOffset = Objects.nonNull(committedOffsetAndMetadata) ? committedOffsetAndMetadata.offset() : null;
-                            LOG.info("Tracking offsets for partition{} starting with committedOffset {}", partition,
-                                    committedOffset);
-                            partitionCommitTrackerMap.put(partitionId, new TopicPartitionCommitTracker(partition, committedOffset));
+                        if (partitionCommitTrackerMap.containsKey(partitionId)) {
+                            final OffsetAndMetadata offsetAndMetadata =
+                                    partitionCommitTrackerMap.get(partitionId).addCompletedOffsets(offsetRange.getOffsets());
+                            updateOffsetsToCommit(partition, offsetAndMetadata);
+                        } else {
+                            LOG.error("!! Commit tracker not found for topic: {} partition: {}", partition.topic(), partitionId);
```

Comment:
I don't think we need these exclamation marks. The log level is `ERROR` which is sufficient here.